### PR TITLE
issue #27 convert to gwasglue2 summaryset inside gwasvcf package

### PR DESCRIPTION
**Pull request by ritarasteiro**
- new feature: creating a gwasglue2 `SummarySet` from a vcf file.
- added new section to guide
 

**NOTE:**

 - There is an error when creating the vignette, related with the `as_tibble` function.
- when investigating further, the error was in `query_gwas,` when `proxies = yes`. I added `name_repair="minimal"` to all the calls of `as_tibble()` in `proxy.r` without it, but it still is given an error. Need to further investigate. 
- @explodecomputer Am I missing something?

_ERRORS:_
- error when running `check()`
``--- re-building ‘guide.Rmd’ using rmarkdown
   /usr/bin/sqlite3
  

   Quitting from lines 271-273 [unnamed-chunk-28] (guide.Rmd)
   Error: processing vignette 'guide.Rmd' failed with diagnostics:
   The `validate` argument of `as_tibble()` was deprecated in tibble 2.0.0
   and is now defunct.
   ℹ Please use the `.name_repair` argument instead.
   --- failed re-building ‘guide.Rmd’
   
   SUMMARY: processing the following file failed:
     ‘guide.Rmd’
   
   Error: Vignette re-building failed.
   Execution halted
Error in `(function (command = NULL, args = character(), error_on_status = TRUE, …`:
! System command 'R' failed``

- error when running code in the vignette:
`Error:
! The `validate` argument of `as_tibble()` was deprecated in tibble 2.0.0 and is now defunct.
ℹ Please use the `.name_repair` argument instead.
Backtrace:
 1. gwasvcf::query_gwas(...)
 6. tibble:::as_tibble.data.frame(., .data, .name_repair = "minimal")
 7. lifecycle::deprecate_stop(...)
 8. lifecycle:::deprecate_stop0(msg)`

### CODE REVIEW

File: R/gwasglue.R
Comment (remlapmot) [inline R/gwasglue.R:17]: Recommend deleting the `.` as it's not needed as you're not putting in a position other than first.

File: R/gwasglue.R
Comment (remlapmot) [inline R/gwasglue.R:18]: Again delete the `.,` because it's not needed.

File: R/proxy.r
Comment (remlapmot) [inline R/proxy.r:59]: Recommend deleting the `.data, ` as it's not needed (and I think this is what's causing the error in building guide.Rmd).

File: R/proxy.r
Comment (remlapmot) [inline R/proxy.r:102]: Recommend deleting the `.data, ` as it's not needed (and I think this is what's causing the error in building guide.Rmd).

**Comment by remlapmot** [comment]
Additionally on line 73 of R/proxy.R, as with last 2 comments recommend amending 
```r
temp <- do.call(rbind, strsplit(ld[["PHASE"]], "")) %>% dplyr::as_tibble(.data, .name_repair="minimal")
```
to simply
```r
temp <- do.call(rbind, strsplit(ld[["PHASE"]], "")) %>% dplyr::as_tibble(.name_repair="minimal")
```

**Comment by remlapmot** [comment]
And the same deletion of `.data, ` is needed on lines 109 and 111 of R/proxy.R as well I think.

**Comment by remlapmot** [comment]
So I think that's 5 instances of `.data, ` to delete in total.

**Comment by remlapmot** [comment]
And I recommend adding yourself to the DESCRIPTION as a ctb at least.

**Comment by remlapmot** [comment]
And I recommend bumping the version number in the DESCRIPTION - we need to get better at remembering to do this in PRs.

**Comment by ritarasteiro** [comment]
Thanks @remlapmot! I made all the changes and also added a NEWS file.

**Comment by remlapmot** [comment]
This looks good to me @ritarasteiro .

My last super minor comment is that is usually format the DESCRIPTION by running `usethis::use_tidy_description()` - which will move gwasglue2 to be in alphabetical order in the Imports list.

Also I think this PR also fixes issue #25. I don't seem to have the rights to add to the Developer pane on this PR page - so I'll put the text below which may hopefully close that issue on merging this.

closes #25

**Comment by chenyangsu** [comment]
Hi MRCIEU team, Would it be possible to merge these pull requests soon? I'm using your package for a project for finding LD proxies but this error still remains.

Error:
! The `validate` argument of `as_tibble()` was deprecated in tibble 2.0.0 and is now defunct.
ℹ Please use the `.name_repair` argument instead.

In addition, I was able to bypass the error by installing the branch with the new changes:
```
library(remotes)
install_github("MRCIEU/gwasvcf@27-convert-to-gwasglue2-summaryset-inside-gwasvcf-package")
```

However, after running 
`query_gwas(outcome_vcf, rsid=input_rsid, proxies="only", bfile=ld_file, build = "GRCh38", tag_kb = 1000, tag_r2=0.8)` there's another error at the align step:

```
Determining searchspace...
Proxy lookup...
Finding proxies...
Taking input= as a system command ('gunzip -c /tmp/RtmpnitEtI/file3098e72a46191.targets.ld.gz') and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.
Found 24 proxies
Extrating proxies...
Identified proxies for 1 of 1 rsids
Aligning...
Error in all_dims[, 1L] : incorrect number of dimensions
```

Would it be possible to solve this issue? Thanks a lot!

**Comment by remlapmot** [comment]
@ritarasteiro / @explodecomputer please merge #31 (if you're happy to) into this branch then merge this into master.

(Note that both of these still don't fix #26 which we'll have to fix after this).

**Review by remlapmot — APPROVED** [review]
This looks good to me now.

**Comment by remlapmot** [comment]
Thanks for merging #31 @ritarasteiro - note the checks here are just failing because the OpenGWAS server is under heavy load.

Also - it's worth noting I don't have any permissions on this repo - so I can't merge your PRs (even if I approve them - I just checked by approving this one, but I didn't obtain the Merge button). I'd be happy to merge this if you or @explodecomputer want to give me permission in the repo (I could be a backup for Gib when he's busy).

**Comment by ritarasteiro** [comment]
@remlapmot  I just added you with maintainer permissions.

The checks appear to fail when trying to install genetics.binaRies in the guide and in ubuntu. I am also having some problems when trying to install this package locally.

**Comment by remlapmot** [comment]
> @remlapmot I just added you with maintainer permissions.

thanks

> The checks appear to fail when trying to install genetics.binaRies in the guide and in ubuntu. I am also having some problems when trying to install this package locally.

Great spot - the genetics.binaRies download fail is because Gib moved that repo from being under his account to under the MRCIEU org. Usually GitHub automatically redirects URLs of moved repos - and if you go to the old repo address it does indeed redirect. However it seems maybe there are some instances where the redirection doesn't occur. So I will send a little fix to genetics.binaRies to use the new-ish MRCIEU address before merging this.

**Comment by ritarasteiro** [comment]
> > @remlapmot I just added you with maintainer permissions.
> 
> thanks
> 
> > The checks appear to fail when trying to install genetics.binaRies in the guide and in ubuntu. I am also having some problems when trying to install this package locally.
> 
> Great spot - the genetics.binaRies download fail is because Gib moved that repo from being under his account to under the MRCIEU org. Usually GitHub automatically redirects URLs of moved repos - and if you go to the old repo address it does indeed redirect. However it seems maybe there are some instances where the redirection doesn't occur. So I will send a little fix to genetics.binaRies to use the new-ish MRCIEU address before merging this.

**Comment by remlapmot** [comment]
@explodecomputer - merging this as it seems good to me.

**Pull request merged**
